emc-metalnx-services/src/main/java/com/emc/metalnx/services/irods/UploadServiceImpl.java

```diff
@@ -64,8 +64,7 @@ public boolean upload(MultipartFile file, String targetPath, boolean computeChec
 
 		logger.info("upload()");
 
-		if (file == null || file.isEmpty() || "".equals(targetPath) || targetPath == null || "".equals(destResc)
-				|| destResc == null) {
+		if (file == null || file.isEmpty() || "".equals(targetPath) || targetPath == null) {
```

Comment:
Swap order of null/empty-string checks on `targetPath`.

Reply:
Good catch.  Done.